src/createRouter.ts

```diff
@@ -130,6 +130,21 @@ export interface RouterFactory extends ComposeFactory<Router, RouterOptions> {
 
 const historyMap = new WeakMap<Router, HistoryManager>();
 
+// istanbul ignore next
+const noop = () => {};
+
+function createDeferral() {
+	// Use noop since TypeScript doesn't know we're assigning cancel and resume in the promise executor.
```

Comment:
this is unfortunate, but i see this is just lifted out of an existing function so fair 👍